src/nccl_ofi_net.c

```diff
@@ -1312,19 +1310,26 @@ static ncclResult_t ofi_listen(int dev, void *handle, void **listenComm)
 	return ret;
 }
 
+struct connection_info {
+	char ep_name[MAX_EP_ADDR];
+	size_t ep_namelen;
+	nccl_ofi_req_t* req;
+	struct connection_info* memory_buffer;
```

Comment:
Is this like a self pointer?

Reply:
memory buffer is pointer to connect_info in sender.  Avoids need for any globals.
For sender side it is self, yes, but for receiver side it is just a pointer to memory which needs to be free'd.